static/js/panels/ProfileForm.tsx

```diff
@@ -11,11 +11,35 @@ import { createProfile } from "../api/profiles";
 import { useQueryClient } from "@tanstack/react-query";
 import { queryKeys } from "../util/queryKeys";
 import SubmitButton from "../buttons/SubmitButton";
+import NetworkSelector from "../components/NetworkSelector";
+import { LxdDevices, LxdDiskDevice, LxdNicDevice } from "../types/device";
+import StorageSelector from "../components/StorageSelector";
+import CpuLimitSelector from "../components/CpuLimitSelector";
+import CloudInitConfig from "../components/CloudInitConfig";
+import { LxdConfigPair } from "../types/config";
+import { LxdProfile } from "../types/profile";
+import MemoryLimitSelector from "../components/MemoryLimitSelector";
+import { CpuLimit, CPU_LIMIT_TYPE, MemoryLimit } from "../types/limits";
+import {
+  DEFAULT_CPU_LIMIT,
+  DEFAULT_DISK_DEVICE,
+  DEFAULT_MEM_LIMIT,
+  DEFAULT_NIC_DEVICE,
+} from "../util/defaults";
 
 const ProfileForm: FC = () => {
   const navigate = useNavigate();
   const notify = useNotification();
   const queryClient = useQueryClient();
+  const [nicDevice, setNicDevice] = useState<LxdNicDevice>(DEFAULT_NIC_DEVICE);
```

Comment:
We can use formik to store the state. It is intended for it and we don't need to define state setter/getter next to it.

Below an example how to use formik for the state of the nicDevice.

```
Subject: [PATCH] fix(instance) simplify instance actions
 - use portal to set actions on instance detail page
 - close instance edit stays on current page
 - extract create snapshot btn
---
Index: static/js/components/NetworkSelector.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/static/js/components/NetworkSelector.tsx b/static/js/components/NetworkSelector.tsx
--- a/static/js/components/NetworkSelector.tsx	(revision 706eb18f5a4b51c0b994db09c7217d8886974d55)
+++ b/static/js/components/NetworkSelector.tsx	(date 1671110410115)
@@ -9,7 +9,7 @@
 interface Props {
   notify: NotificationHelper;
   nicDevice: LxdNicDevice;
-  setNicDevice: Dispatch<SetStateAction<LxdNicDevice>>;
+  setNicDevice: (nic: LxdNicDevice) => void;
 }
 
 const NetworkSelector: FC<Props> = ({ notify, nicDevice, setNicDevice }) => {
Index: static/js/panels/ProfileForm.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/static/js/panels/ProfileForm.tsx b/static/js/panels/ProfileForm.tsx
--- a/static/js/panels/ProfileForm.tsx	(revision 706eb18f5a4b51c0b994db09c7217d8886974d55)
+++ b/static/js/panels/ProfileForm.tsx	(date 1671110547987)
@@ -12,7 +12,7 @@
 import { queryKeys } from "../util/queryKeys";
 import SubmitButton from "../buttons/SubmitButton";
 import NetworkSelector from "../components/NetworkSelector";
-import { LxdDevices, LxdDiskDevice, LxdNicDevice } from "../types/device";
+import { LxdDevices, LxdDiskDevice } from "../types/device";
 import StorageSelector from "../components/StorageSelector";
 import CpuLimitSelector from "../components/CpuLimitSelector";
 import CloudInitConfig from "../components/CloudInitConfig";
@@ -31,7 +31,6 @@
   const navigate = useNavigate();
   const notify = useNotification();
   const queryClient = useQueryClient();
-  const [nicDevice, setNicDevice] = useState<LxdNicDevice>(DEFAULT_NIC_DEVICE);
   const [rootDiskDevice, setRootDiskDevice] =
     useState<LxdDiskDevice>(DEFAULT_DISK_DEVICE);
   const [cpuLimit, setCpuLimit] = useState<CpuLimit>(DEFAULT_CPU_LIMIT);
@@ -49,14 +48,16 @@
     initialValues: {
       name: "",
       description: "",
+      cpuLimit: DEFAULT_CPU_LIMIT,
+      nicDevice: DEFAULT_NIC_DEVICE,
     },
     validationSchema: ProfileSchema,
-    onSubmit: (values) => {
+    onSubmit: ({ name, description, cpuLimit, nicDevice }) => {
       // create LxdDevices object
       const devicesObj: LxdDevices = {};
       if (nicDevice.network) {
-        devicesObj[nicDevice.name === "" ? nicDevice.network : nicDevice.name] =
-          nicDevice;
+        const name = nicDevice.name === "" ? nicDevice.network : nicDevice.name;
+        devicesObj[name] = nicDevice;
       }
       if (rootDiskDevice.pool) {
         devicesObj.root = rootDiskDevice;
@@ -111,9 +112,9 @@
       // build final LxdProfile object with all the data
       const profile: LxdProfile = {
         config: configObj,
-        description: values.description,
+        description: description,
         devices: devicesObj,
-        name: values.name,
+        name: name,
       };
 
       createProfile(profile)
@@ -170,8 +171,10 @@
                     content: (
                       <NetworkSelector
                         notify={notify}
-                        nicDevice={nicDevice}
-                        setNicDevice={setNicDevice}
+                        nicDevice={formik.values.nicDevice}
+                        setNicDevice={(v) =>
+                          void formik.setFieldValue("nicDevice", v)
+                        }
                       />
                     ),
                     title: "Network",
```